### PR TITLE
Fixes deprecation warning for boost::timer

### PR DESCRIPTION
**Pull request by sarahec**
The code was using boost::timer version 1 while Boost has moved to a new API. The new timer API is very similar to the one introduced in C++ 11, so I just went with C++11.

Notes:

1. To reset the timer,  just allocate a new one.
2. To calculate elapsed time,  subtract the timer from now (a new timer)
3. Then cast the result to the desired granularity. This code is casting to seconds to match the original.

### CODE REVIEW

File: openasip/src/applibs/Scheduler/Algorithms/ProgramPass.cc [hunk 1]
```diff
@@ -72,13 +73,12 @@ void
 ProgramPass::executeProcedurePass(
     TTAProgram::Program& program, const TTAMachine::Machine& targetMachine,
     ProcedurePass& procedurePass) {
-    boost::timer totalTime;
+  auto totalTimeStart = std::chrono::steady_clock::now();
```
Comment (pjaaskel) [inline openasip/src/applibs/Scheduler/Algorithms/ProgramPass.cc:76]: Is there an indent problem here? Can you try if clang-format cleans this up correctly? 4 spaces indents.

**Review by pjaaskel — CHANGES_REQUESTED** [review]
Thanks a lot for working on this. I think many of the Boost libs are now found in C++, it's good to drop these dependencies off one-by-one.

However, I think there are some stylistic issues, can you try if the clang formatter script works here, they are under tools/scripts, but might need a bit of tweaking as they don't get all of the style right. src/base/mach/* is a good reference for the original TCE style. 4 spaces no tabs indentation, nearly stroustrup style. Thanks again!

File: openasip/src/applibs/Scheduler/Algorithms/ProgramPass.cc (hunk 1 above)
Comment (sarahec) [reply openasip/src/applibs/Scheduler/Algorithms/ProgramPass.cc]: Fixed by hand.

I need to build a clang formatting file to match your style. clang-format has a tendency to make widespread changes to your code. (See: the rearranged headers in ControlDependanceGraph. I didn't worry about putting these back in the original order as it compiles and tests properly.)

**Comment by sarahec** [comment]
Are there additional stylistic issues you see? I reviewed the rest of the diffs and they look good.

**Comment by sarahec** [comment]
The above changes update the formatting scripts to work today and then run them across the files. Anything else you'd like?

**Review by pjaaskel — APPROVED** [review]
Thanks! Looks good now. Will test&push. Well, perhaps you could add this internal update (which also drops one external dependence and makes the code base more portable) to CHANGES, but it can be done in the next merge from you.

**Pull request merged**